qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+        This feels somewhat hacky/arbitrary.
+        """
+        start = time.time()
+
+        # map measurement subsystems from labels to correct index
+        if self.subsystem_labels:
+            new_measurement_subsystems_list = []
+            for measurement_subsystems in measurement_subsystems_list:
+                new_measurement_subsystems = []
+                for subsystem in measurement_subsystems:
+                    if subsystem in self.subsystem_labels:
+                        new_measurement_subsystems.append(self.subsystem_labels.index(subsystem))
+                    else:
+                        raise QiskitError(f"Attempted to measure subsystem {subsystem}, but it is not in subsystem_list.")
+                new_measurement_subsystems_list.append(new_measurement_subsystems)
+
+            measurement_subsystems_list = new_measurement_subsystems_list
+
+        solver_results = self.solver.solve(t_span=t_span, y0=y0, signals=schedules, **solver_options)
+
+        # construct counts for each experiment
+        ##############################################################################################
+        # Change to if statement depending on types of outputs, e.g. counts vs IQ data
+        outputs = []
+        for ts, result, measurement_subsystems in zip(t_span, solver_results, measurement_subsystems_list):
+            yf = result.y[-1]
+
+            # Put state in dressed basis and sample counts
+            if isinstance(yf, Statevector):
+                yf = np.array(self.solver.model.rotating_frame.state_out_of_frame(t=ts[-1], y=yf))
+                yf = self._dressed_states_adjoint @ yf
+                yf = Statevector(yf, dims=self.subsystem_dims)
+
+                if normalize_states:
+                    yf = yf / np.linalg.norm(yf.data)
+            elif isinstance(yf, DensityMatrix):
+                yf = np.array(self.solver.model.rotating_frame.operator_out_of_frame(t=ts[-1], y=yf))
+                yf = self._dressed_states_adjoint @ yf @ self.dressed_states
+                yf = DensityMatrix(yf, dims=self.subsystem_dims)
+
+                if normalize_states:
+                    yf = yf / np.diag(yf.data).sum()
+
+            outputs.append({'counts': yf.sample_counts(shots=shots, qargs=measurement_subsystems)})
+
+        results_list = []
+        for schedule, output in zip(schedules, outputs):
+            results_list.append(
+                Result(
+                    backend_name=self.name,
+                    backend_version=self.backend_version,
+                    qobj_id=None, # Should we subclass result or something because qobj_id doesn't exist?
+                    job_id=job_id,
+                    success=True,
+                    results=output,
+                    date=datetime.datetime.now().isoformat()
+                )
+            )
+
+        return results_list
+
+    @property
+    def meas_map(self) -> List[List[int]]:
```

Comment:
Since we don't need to multiplex a simulator I think this should just be a list like `[[0], ..., [N-1]]`. You could also just leave it as `NotImplementedError` from the base class